src/EventStore.Core/Services/Storage/IndexCommitterService.cs

```diff
@@ -440,6 +447,41 @@ public bool IsReplicated(int commitCount)
                 }
             }
         }
+
+        private Task _mergeIndexesTask;
+
+        public void Handle(ClientMessage.MergeIndexes message)
+        {
+            if (!IsAllowed(message.User, message.CorrelationId, message.Envelope)) return;
+            if (_mergeIndexesTask != null && _mergeIndexesTask.Status == TaskStatus.Running)
+            {
+                Log.Info("Index Merge Operation already running...");
+                MakeReplyForMergeIndexes(message);
+                return;
+            }
+            _mergeIndexesTask = _tableIndex.MergeIndexes().ContinueWith(t =>
+            {
+                _tableIndex.Scavenge(_logManager.CreateLog(), default(CancellationToken));
```

Comment:
@riccardone Index merges automatically remove deleted entries in the PTables being merged. The intent of the `admin/mergeindexes` endpoint is only to merge pending PTables. I don't think it's a good idea to scavenge the index here since it's an operation that needs to go over the whole database (all chunk files and PTabls) and will be very time consuming/disk intensive. Besides, DB/Index scavenges can be launched manually by the user and if we want a user to be able to launch only an index scavenge (and not DB scavenge), a better place to add it would be in the `/admin/scavenge` endpoint in another PR.

Reply:
It's already done and merged from the Scavenge_Indexes PR. When you scavenge database chunks using /admin/scavenge you also scavenge the indexes. That scavenge operation was running automatically before that Scavenge_Indexes PR as part of the index merge. Now that we are making the operation on the indexes controllable, I think that it make sense run Merge and Scavenge together on indexes.

Reply:
`That scavenge operation was running automatically before that Scavenge_Indexes PR as part of the index merge`

@riccardone Only the PTables being merged were being scavenged before the Scavenge_Indexes PR. e.g. if you had two 1M PTables being merged into a single 2M PTable and there were X deleted entries, the merge would result in a 2M-X PTable. All other PTables would remain unaffected and would not be scavenged.